src/threading/SerialTypes.h

```diff
@@ -47,6 +47,20 @@ struct Field {
 		delete [] secondary_name;
 		}
 
+	Field& operator=(const Field& other)
+		{
+		if ( this != &other )
+			{
+			name = other.name ? util::copy_string(other.name) : nullptr;
+			secondary_name = other.secondary_name ? util::copy_string(other.secondary_name) : nullptr;
```

Comment:
Looks like these should first `delete []` to avoid leaking.  Also a bit simpler to rely on `copy_string` to perform the null check.